### PR TITLE
Make EIA parser raise errors

### PR DESCRIPTION
**Pull request by pierresegonne**
EIA parsers have been down for a while.
The EIA endpoint returns and error, explaining that the API key is not valid anymore, but the error is not captured and does not lead to an exception being raised. 

This PR addresses this lack.

### CODE REVIEW

File: parsers/EIA.py [hunk 1]
```diff
@@ -361,133 +367,163 @@ def fetch_production_mix(zone_key, session=None, target_datetime=None, logger=No
...
 
+    eia_error_message = None
+    # check nested key with forgiveness
+    try:
+        eia_error_message = raw_data["data"]["error"]
+    except KeyError:
+        pass
+    if eia_error_message:
+        logger.error(f"EIA error: {eia_error_message}")
+        return []
```
Comment (madsnedergaard) [inline parsers/EIA.py:498]: Hmm, couldn't you do it like this instead?

```python
    try:
        eia_error_message = raw_data["data"]["error"]
        logger.error(f"EIA error: {eia_error_message}")
        return []
    except KeyError:
        pass
```

Or maybe even this? Not sure what's clearer:
```python
    eia_error_message = raw_data.get('data', {}).get('error')
    if eia_error_message: 
        logger.error(f"EIA error: {eia_error_message}")
        return []


```

Comment (pierresegonne) [reply parsers/EIA.py]: Good suggestion! Didn't know about the `.get` trick ;)

**Review by madsnedergaard — APPROVED** [review]
Sorry, thought I approved it again yesterday!

File: parsers/EIA.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 https://www.eia.gov/opendata/register.php
 """
 import datetime
+from logging import error
```
Comment (skovhus) [inline parsers/EIA.py:12]: Do we use this import?

Comment (pierresegonne) [reply parsers/EIA.py]: removed it :)

**Pull request merged**